tools/ami-creator/userdata/mxnetlinux_gpu_ubuntu_2004.txt

```diff
@@ -0,0 +1,81 @@
+#cloud-config
+
+apt_reboot_if_required: false
+package_update: true
+package_upgrade: true
+apt:
+  sources:
+    nvidia-cuda-drivers.list:
+      source: |
+        deb https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64 /
+      keyid: 7FA2AF80
+    docker.list:
+      source: deb [arch=amd64] https://download.docker.com/linux/ubuntu $RELEASE stable
+      keyid: 0EBFCD88
+    nvidia-container-runtime.list:
+      source: |
+        deb https://nvidia.github.io/libnvidia-container/ubuntu20.04/$(ARCH) /
+        deb https://nvidia.github.io/nvidia-container-runtime/ubuntu20.04/$(ARCH) /
+        deb https://nvidia.github.io/nvidia-docker/ubuntu20.04/$(ARCH) /
+      keyid: F796ECB0
+
+apt_get_command: ["apt-get", "--option=Dpkg::Options::=--force-confold", "--option=Dpkg::options::=--force-unsafe-io", "--assume-yes", "--quiet", "--install-recommends"]
+
+packages:
+  - wget
+  - curl
+  - openjdk-8-jre
+  - git
+  - python3
+  - python3-pip
+  - cuda-drivers
+  - apt-transport-https
+  - ca-certificates
+  - software-properties-common
+  - docker-ce
+  - nvidia-container-toolkit
+  - nvidia-container-runtime
+
+write_files:
+  - path: /etc/cron.d/jenkins-start-slave
+    content: |
+      @reboot jenkins_slave /home/jenkins_slave/scripts/launch-autoconnect.sh
+  - path: /home/jenkins_slave/scripts/launch-autoconnect.sh
+    content: |
+      #!/bin/sh
+      set -ex
+      sleep 300
```

Comment:
Are you really sure about that? The entire auto scaling is built on the premise that the instance quickly starts up. Just introducing this sleep would be really harmful for the overall performance. Also, there are no package installations happening when the slave starts - that kind of stuff happens during AMI creation.